chore: Set isolation level to READ COMMITTED for testing et al.

john-bodley opened the pull request:
<!---
Please write the PR title following the conventions at https://www.conventionalcommits.org/en/v1.0.0/
Example:
fix(dashboard): load charts correctly
-->

### SUMMARY

The default isolation level for PostgreSQL is `READ COMMITTED` whereas for MySQL it is `REPEATABLE READ`. Ideally we would strive to have these the same (for consistency reasons) and the former is likely preferred given the async nature of Superset (especially SQL Lab) where the backend is polling to determine whether a query has been cancelled from another Flask-SQLAlchemy session.

This PR:

1. Adds a comment in `config.py` to try to persuade institutions to set the isolation level. It's actually hard/difficult to impose this given that the default config is configured for SQLite which doesn't support `READ COMMITTED`.
2. Updates the test configs to ensure that—for MySQL and PostgreSQL—the isolation level is `READ COMMITTED`. 

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
<!--- Skip this if not applicable -->

### TESTING INSTRUCTIONS

CI.

### ADDITIONAL INFORMATION
<!--- Check any relevant boxes with "x" -->
<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->
- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

john-bodley committed:
chore: Set isolation level as READ COMMITTED

john-bodley commented on tests/integration_tests/superset_test_config.py:
```diff
@@ -52,12 +54,15 @@
     UPLOAD_FOLDER = os.environ["UPLOAD_FOLDER"]  # noqa: F405
 
-if "sqlite" in SQLALCHEMY_DATABASE_URI:
+if make_url(SQLALCHEMY_DATABASE_URI).get_backend_name() == "sqlite":
```
Safer than doing a string match.

michael-s-molina commented:
> Set isolation level to READ COMMITTED 

@john-bodley looking at the changed files I was not able to see where you set the isolation level to READ COMMITTED in this PR. Did you forget to commit something?

john-bodley commented:
@michael-s-molina I updated the PR title. It's actually hard to set this globally given that not all of our backends support the `READ COMMITTED`. I did add a comment in the `config.py` and updated the test configs so both MySQL and PostgreSQL will have similar behavior.

sadpandajoe commented on superset/config.py:
```diff
@@ -196,6 +196,13 @@ def _try_json_readsha(filepath: str, length: int) -> str | None:
+# isolation level is READ COMMITTED. All backends should use READ COMMITTED (or similar)
+# to help ensure consistent behavior.
+SQLALCHEMY_ENGINE_OPTIONS = {
+    "isolation_level": "SERIALIZABLE",  # SQLite does not support READ COMMITTED.
```
@john-bodley Reading this it looks like everything will be `SERIALIZABLE` right? we're overriding it in the tests but are we overriding it as `READ COMMITTED` anywhere else for the other databases?

mistercrunch replied:
@sadpandajoe followed the rabbit hole here, and I think you're right, this will bump all envs who don't set their SQLALCHEMY_ENGINE_OPTIONS to a higher isolation level.

mistercrunch replied:
How about this draft? https://github.com/apache/superset/pull/30174
